src/settings.ts

```diff
@@ -0,0 +1,122 @@
+import { WorkspaceConfiguration, workspace, window, commands } from "vscode";
+import { ScopeInfo } from "./extension";
+
+let vmArgsCache;
+let ignoreAutoCloseTags = false;
+let ignoreVMArgs = false;
+let oldConfig: WorkspaceConfiguration = getXMLConfiguration();
+
+
+
+export function getXMLConfiguration(): WorkspaceConfiguration {
+  return workspace.getConfiguration("xml");
+}
+
+export function onConfigurationChange() {
+  if(!ignoreAutoCloseTags) {
+    verifyAutoClosing();
+  }
+  
+  !ignoreVMArgs ? verifyVMArgs(): undefined;
+
+  workspace.onDidChangeConfiguration(params => {
```

Comment:
this seems wrong. This needs to be registered once. Calling it from inside middleware.workspace.didChangeConfiguration doesn't seem right. There must be a way to get the change event directly from middleware.workspace.didChangeConfiguration